lib/OpenQA/Task/Job/Limit.pm

```diff
@@ -24,18 +24,27 @@ sub register {
     $app->minion->add_task(limit_results_and_logs => sub { _limit($app, @_) });
 }
 
+sub _limit_group {
```

Comment:
This could also be part of `JobGroups.pm`. At least it is only concerned about a certain group. And you create a fake group object for groupless jobs anyways.